### PR TITLE
(PDB-3096) log elapsed time with command processing

### PR DESCRIPTION
**Pull request by wkalt**
Change command processing logging to include elapsed ms. This will be
useful to support for correlating slow processing to specific nodes.

### CODE REVIEW

**Comment by wkalt** [comment]
example:

```
2017-03-24 15:36:02,205 INFO  [p.p.command] [568-1490394910578] [5 ms] 'replace facts' command processed for host-0
2017-03-24 15:36:02,207 WARN  [p.p.s.storage] Not replacing catalog for certname host-5 because local data is newer.
2017-03-24 15:36:02,208 INFO  [p.p.command] [567-1490394910553] [41 ms] 'replace catalog' command processed for host-5
2017-03-24 15:36:02,256 INFO  [p.p.command] [564-1490394910525] [131 ms] 'store report' puppet v3.7.1 command processed for host-7
2017-03-24 15:36:02,267 INFO  [p.p.command] [566-1490394910549] [108 ms] 'store report' puppet v3.7.2 command processed for host-5
```

**Comment by mullr** [comment]
@Sharpie How does that format look?

**Comment by Sharpie** [comment]
Format looks good to me 👍

File: src/puppetlabs/puppetdb/command.clj [hunk 1]
```diff
@@ -266,42 +266,45 @@
...
 
 (defn replace-catalog*
-  [{:keys [certname version id received payload]} db]
+  [{:keys [certname version id received payload]} start-time db]
   (let [{producer-timestamp :producer_timestamp :as catalog} payload]
     (jdbc/with-transacted-connection' db :repeatable-read
       (scf-storage/maybe-activate-node! certname producer-timestamp)
       (scf-storage/replace-catalog! catalog received))
-    (log/info (trs "[{0}-{1}] ''{2}'' command processed for {3}"
+    (log/info (trs "[{0}-{1}] [{2} ms] ''{3}'' command processed for {4}"
```
Comment (rbrw) [inline src/puppetlabs/puppetdb/command.clj]: Looks like this is formatting with thousands separators - not sure we want that (e.g. 1,2341 ms)?  It'll make parsing harder if anyone's likely to try to parse this (locale dependent).

Comment (rbrw) [inline src/puppetlabs/puppetdb/command.clj]: Alternately, I suppose you could argue that's what structured logging's for...

File: src/puppetlabs/puppetdb/command.clj [hunk 2]
```diff
@@ -266,42 +266,45 @@
...
 (defn replace-facts*
-  [{:keys [payload id received] :as command} db]
+  [{:keys [payload id received] :as command} start-time db]
   (let [{:keys [certname values] :as fact-data} payload
         producer-timestamp (:producer_timestamp fact-data)]
     (jdbc/with-transacted-connection' db :repeatable-read
       (scf-storage/maybe-activate-node! certname producer-timestamp)
       (scf-storage/replace-facts! fact-data))
-    (log/info (trs "[{0}-{1}] ''{2}'' command processed for {3}"
+    (log/info (trs "[{0}-{1}] [{2} ms] ''{3}'' command processed for {4}"
```
Comment (rbrw) [inline src/puppetlabs/puppetdb/command.clj]: If this is the same code as above, and below, wonder if a trivial helper function might be useful.

**Comment by rbrw** [comment]
Once we resolve the "1,234 ms" question, looks good.

File: src/puppetlabs/puppetdb/command.clj [hunk 3]
```diff
@@ -382,14 +387,15 @@
 
 (defn process-command!
   "Takes a command object and processes it to completion. Dispatch is
-  based on the command's name and version information"
+   based on the command's name and version information"
```
Comment (rbrw) [inline src/puppetlabs/puppetdb/command.clj]: Possibly extraneous.

File: src/puppetlabs/puppetdb/command.clj (hunk 1 above)
Comment (Sharpie) [inline src/puppetlabs/puppetdb/command.clj]: Oh yeah... locale specific number formats are not good in this case as this information will definitely be parsed using a script or tool like logstash and having to fiddle about with where the comma goes will slow troubleshooting down for no real benefit.

**Comment by mullr** [comment]
@rbrw I factored out a logging function and fixed the formatting; can you take a look and merge when ready?

**Pull request merged**